crates/core/src/config.rs

```diff
@@ -84,25 +84,38 @@ impl Default for Config {
 }
 
 impl Config {
-    pub fn from_env() -> Self {
-        Self {
+    pub fn from_env() -> Result<Self> {
+        fn require_env_var(var_key: &'static str) -> Result<String> {
+            std::env::var(var_key).map_err(|_| eyre!("The \"{}\" env var must be set or a configuration file must be specified", var_key))
+        }
+
+        Ok(Self {
             network: Network::from_str(
                 &std::env::var("NETWORK").unwrap_or_default(),
             )
             .unwrap_or(Network::MAINNET),
-            eth_execution_rpc: std::env::var("ETH_EXECUTION_RPC")
-                .unwrap_or_default(),
-            starknet_rpc: std::env::var("STARKNET_RPC").unwrap_or_default(),
+            eth_execution_rpc: require_env_var("ETH_EXECUTION_RPC")?,
+            starknet_rpc: require_env_var("STARKNET_RPC")?,
             data_dir: PathBuf::from(
                 std::env::var("DATA_DIR").unwrap_or_default(),
```

Comment:
Maybe for separate PR: sanity check that directory exists and is writable seems reasonable here

Reply:
https://github.com/eigerco/beerus/issues/632